### PR TITLE
AlamedaService Set Alameda-Component's Log and InfluxDB,Grafana Storage PersistentVolumeClaim  resolved #4

### LINKED ISSUE #4: Support Custom PersistentVolumeClaim definition
Reference:
https://github.com/coreos/etcd-operator/blob/master/doc/user/spec_examples.md#custom-persistentvolumeclaim-definition

https://github.com/coreos/etcd-operator/blob/master/pkg/cluster/cluster.go#L372

https://github.com/coreos/etcd-operator/blob/master/pkg/util/k8sutil/k8sutil.go#L285

### PR DESCRIPTION
**Pull request by ekko771**

### CODE REVIEW

**Comment by mamafun** [comment]
Suggest to reorganize alamedaservice structure as follows:
The idea is to keep the configure simple (as global setting)
and also have some flexibility for each component.
@tsungming @ekko771 @rax845163 @kfdev170707 any comments?

Note: each Alameda components should store their data in /var/log/alameda/<component name>

```
apiVersion: federatorai.containers.ai/v1alpha1
kind: AlamedaService
metadata:
  name: my-alamedaservice
  namespace: alam
spec:
  enableexecution: true
  enablegui: true
  imageHub: quay.io/prophetstor ## for alameda components. (exclude grafana and influxdb)
  imageBase: ubi                ## for alameda components. (exclude grafana and influxdb)
  version: v0.3.7               ## for alameda components. (exclude grafana and influxdb)
  prometheusservice: http://prometheus-prometheus-oper-prometheus.monitoring:9090
  storage:                      ## see following details for where it is used in each component
    type: ephemeral | pvc       ## default is ephemeral means emptyDir. pvc means persistent volume claim
    size: 10Gi                  ## mandatory if type=pvc;
    class: ""                   ## mandatory if type=pvc;
    accessModes:                ## default is ["ReadWriteOnce"]. (this is a list)

# following is more detail configures for each components and overwrite the global config
  alameda-operator:
    image: quay.io/prophetstor/alameda-operator-rhel
    version: v0.3.7
    storage:                    ## for /var/lib/alameda/operator
      type: ephemeral | pvc     ## default is ephemeral means emptyDir. pvc means persistent volume claim
      size: 10Gi                ## mandatory if type=pvc;
      class: ""                 ## mandatory if type=pvc;
      accessModes:              ## default is ["ReadWriteOnce"]. (this is a list)

  alameda-datahub:
    image: quay.io/prophetstor/alameda-operator-rhel
    version: v0.3.7
    storage:                    ## for /var/lib/alameda/datahub
      type: ephemeral | pvc     ## default is ephemeral means emptyDir. pvc means persistent volume claim
      size: 10Gi                ## mandatory if type=pvc;
      class: ""                 ## mandatory if type=pvc;
      accessModes:              ## default is ["ReadWriteOnce"]. (this is a list)
	

  alameda-ai:
    image: quay.io/prophetstor/alameda-operator-rhel
    version: v0.3.7
    storage:                    ## for /var/lib/alameda/ai
      type: ephemeral | pvc     ## default is ephemeral means emptyDir. pvc means persistent volume claim
      size: 10Gi                ## mandatory if type=pvc;
      class: ""                 ## mandatory if type=pvc;
      accessModes:              ## default is ["ReadWriteOnce"]. (this is a list)

  alameda-evictioner:
    image: quay.io/prophetstor/alameda-operator-rhel
    version: v0.3.7
    storage:                    ## for /var/lib/alameda/evictioner
      type: ephemeral | pvc     ## default is ephemeral means emptyDir. pvc means persistent volume claim
      size: 10Gi                ## mandatory if type=pvc;
      class: ""                 ## mandatory if type=pvc;
      accessModes:              ## default is ["ReadWriteOnce"]. (this is a list)

  alameda-admission-controller:
    image: quay.io/prophetstor/alameda-operator-rhel
    version: v0.3.7
    storage:                    ## for /var/lib/alameda/admission-controller
      type: ephemeral | pvc     ## default is ephemeral means emptyDir. pvc means persistent volume claim
      size: 10Gi                ## mandatory if type=pvc;
      class: ""                 ## mandatory if type=pvc;
      accessModes:              ## default is ["ReadWriteOnce"]. (this is a list)

  alameda-grafana:
    image: quay.io/prophetstor/alameda-operator-rhel
    version: v0.3.7
    storage:                    ## for /var/lib/grafana
      type: ephemeral | pvc     ## default is ephemeral means emptyDir. pvc means persistent volume claim
      size: 10Gi                ## mandatory if type=pvc;
      class: ""                 ## mandatory if type=pvc;
      accessModes:              ## default is ["ReadWriteOnce"]. (this is a list)

  alameda-influxdb:
    image: quay.io/prophetstor/alameda-operator-rhel
    version: v0.3.7
    storage:                    ## for /var/lib/influxdb
      type: ephemeral | pvc     ## default is ephemeral means emptyDir. pvc means persistent volume claim
      size: 10Gi                ## mandatory if type=pvc;
      class: ""                 ## mandatory if type=pvc;
      accessModes:              ## default is ["ReadWriteOnce"]. (this is a list)
  ```

**Comment by rax845163** [comment]
@mamafun 
I suggest reorganizing the storage schema to the following configuration.
```
  storages:                     
  - usage: data         
    type: ephemeral              
  - usage: log         
    type: pvc     
    size: 10Gi                 
    class: "awsEBS"                   
    accessMode: "ReadWriteOnce"  ## a single value to represent which access mode should the pvc adopt, cause a volume can only be mounted using one access mode at a time, even if it supports many. (Refrence document https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes)
```

**Comment by tsungming** [comment]
Please follow #102 to discuss the new structure

File: pkg/apis/federatorai/v1alpha1/alamedaservice_types.go [hunk 1]
```diff
@@ -14,11 +15,60 @@ type AlamedaServiceSpec struct {
...
-	EnableGUI             bool   `json:"enablegui"`
-	Version               string `json:"version"`
-	PrometheusService     string `json:"prometheusservice"`
-	PersistentVolumeClaim string `json:"persistentvolumeclaim"`
+	EnableExecution        bool                             `json:"enableexecution"`
+	EnableGUI              bool                             `json:"enablegui"`
+	Version                string                           `json:"version"`
+	PrometheusService      string                           `json:"prometheusservice"`
+	PersistentVolumeClaim  string                           `json:"persistentvolumeclaim"`
+	InfluxdbPVCSet         AlamedaServiceSpecInfluxdbPVCSet `json:"influxdbpvcset"`
```
Comment (mamafun) [inline pkg/apis/federatorai/v1alpha1/alamedaservice_types.go]: structure _AlamedaServiceSpecInfluxdbPVCSet_  and _AlamedaServiceSpecGrafanaPVCSet_  looks the same. why we need both?

File: pkg/apis/federatorai/v1alpha1/alamedaservice_types.go [hunk 2]
```diff
@@ -14,11 +15,60 @@ type AlamedaServiceSpec struct {
...
+	GrafanaPVCSet          AlamedaServiceSpecGrafanaPVCSet  `json:"grafanapvcset"`
+	AlamedaAILog           AlamedaAILog                     `json:"alamedaailog"`
```
Comment (mamafun) [inline pkg/apis/federatorai/v1alpha1/alamedaservice_types.go]: structure AlamedaAILog, AlamedaOperatorLoglooks, AlamedaDatahubLog, AlamedaEvictionerLog and AdmissionControllerLog looks the same. why we need to separate them in 5 struct?

**Pull request closed without merging**